src/pages/Scholarships.jsx

```diff
@@ -48,6 +48,7 @@ function ScholarshipsPage() {
   const [scholarships, setScholarships] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState();
+  const [lastDoc, setLastDoc] = useState(null);
```

Comment:
Instead of a state variable I suggest just deriving this from `scholarships`' last item.